src/test/java/org/folio/rest/impl/SamlAPITest.java

```diff
@@ -498,6 +505,33 @@ public void putConfigurationEndpoint(TestContext context) {
       .body(matchesJsonSchemaInClasspath("ramls/schemas/SamlConfig.json"));
   }
 
+   @Test
+  public void putConfigurationWithIdmXml(TestContext context) throws IOException {
+    SamlConfigRequest samlConfigRequest = new SamlConfigRequest()
+      .withIdpUrl(URI.create("http://localhost:" + MOCK_PORT + "/xml"))
+      .withSamlAttribute("UserID")
+      .withSamlBinding(SamlConfigRequest.SamlBinding.POST)
+      .withUserProperty("externalSystemId")
+      .withIdmXml(IOUtils.toString(Objects
+        .requireNonNull(getClass().getClassLoader()
+          .getResourceAsStream("mock_idm.xml")), StandardCharsets.UTF_8))
+      .withOkapiUrl(URI.create("http://localhost:9130"));
+
+    String jsonString = Json.encode(samlConfigRequest);
+
+    // PUT
+    given()
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .header(OKAPI_URL_HEADER)
+      .header(JSON_CONTENT_TYPE_HEADER)
+      .body(jsonString)
+      .put("/saml/configuration")
+      .then()
+      .statusCode(200)
+      .body(matchesJsonSchemaInClasspath("ramls/schemas/SamlConfig.json"));
```

Comment:
Can you extend the tests by sending a response to mod-login-saml and check that a response signed with correct certificate is accepted and a response signed with wrong certificate is rejected?

Reply:
Is there any example?

Reply:
You might try some reference environment with SSO enabled and copy&paste the response from the browser developer tools.